### PR TITLE
feat: add file processor component

### PR DESCRIPTION
**Pull request by Hitesh-Kandpal**
### Pre-requisites

- [x] I have gone through the Contributing guidelines for [Submitting a Pull Request (PR)](../../CONTRIBUTING.md#submitting-a-pull-request-pr) and ensured that this is not a duplicate PR.
- [x] I have performed preliminary testing to ensure that any existing features are not impacted and any new features are working as expected.

### PR Details

PR details have been updated as per the given format (see below)

- [x] PR title adheres to the format specified in guidelines (e.g., `feat: add admin login page`)
- [x] Description has been added
- [x] Related changes have been added (optional)
- [x] Screenshots have been added (optional)
- [ ] Pending actions have been added (optional)
- [ ] Any other additional notes have been added (optional)

### Additional Information

- [x] Appropriate label(s) have been added (`ready for review` should be added if the PR is ready to be reviewed)
- [x] Assignee(s) and reviewer(s) have been added (optional)

---

**Description:**

added a file processor component to show status of files and give a functionality to download it

**Related changes:**

1. added a file-processor component
2. change folder structuring a little bit, stored transcript analyzer component in a folder inside features folder
3. did the necessary changes in app.routing and app.module
4. create a interface for storing job details

**Screenshots:**

lint and build result:
Screenshot 2024-05-03 at 12 35 24 PM

https://github.com/OsmosysSoftware/osm-transcript-summarizer/assets/168435416/c26324ce-c94f-441e-bb61-0d42f48eb1ea

**Pending actions:**

N/A

**Additional notes:**

transcript-analyzer code is same as before I just stored it in a new folder

## Summary by CodeRabbit

- **New Features**
  - Introduced a new `FileProcessorComponent` for displaying and managing job details, including file downloads.
  - Added a `TranscriptAnalyzerComponent` for uploading and analyzing transcripts.
  - Added a job table structure in the `file-processor.component.html` file.

- **Enhancements**
  - Updated routing to improve navigation within the application.
  - Added `downloadFile(summary: string, fileName: string): void` method to the `FileProcessorComponent` class.
  - Added `summarizeTranscript(): void` method in class `TranscriptAnalyzerComponent`.

- **User Interface**
  - Added new user interfaces for both `FileProcessorComponent` and `TranscriptAnalyzerComponent` with corresponding styles.
  - Introduced styling for the job table in the `file-processor.component.scss` file and for the transcript analyzer feature in the `transcript-analyzer.component.scss` file.

- **Documentation**
  - Updated documentation to reflect new component functionalities and interfaces.
  
- **Dependencies**
  - Added `"file-saver": "^2.0.5"` and `"@types/file-saver": "^2.0.7"` dependencies in `package.json` for file saving functionality.
  
- **Refactor**
  - Updated the import path for `TranscriptAnalyzerComponent` in the Angular application routing module.
  - Added a new import statement for `TagModule` from 'primeng/tag' in the `primeng.module.ts` file.

### CODE REVIEW

File: apps/portal/src/app/features/file-processor/file-processor.component.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+  constructor(private http: HttpClient) {}
+
+  ngOnInit(): void {
+    // Dummy data for demonstration
+    const dummyJobs: JobDetails[] = [
+      {
+        id: 1,
+        fileName: 'meeting_minutes.txt',
+        status: 'IN PROGRESS',
```
Comment (xixas) [inline apps/portal/src/app/features/file-processor/file-processor.component.ts:21]: Status should come from Enum value.

File: apps/portal/src/app/features/file-processor/file-processor.component.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+8. Consider multi-language support and terms of service agreement for global user base and legal compliance.`,
+      },
+    ];
+
+    // later data will be fetched from api in here
+    this.jobs = dummyJobs;
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  downloadFile(summary: string, fileName: string): void {
```
Comment (xixas) [inline apps/portal/src/app/features/file-processor/file-processor.component.ts:101]: Use the file-saver package instead like metnioned in task(make sure to add types package too).

File: apps/portal/src/app/features/transcript-analyzer/transcript-analyzer.component.scss [hunk 3]
```diff
@@ -0,0 +1,25 @@
+@import 'primeflex/primeflex.scss';
```
Comment (xixas) [inline apps/portal/src/app/features/transcript-analyzer/transcript-analyzer.component.scss:1]: I don't remember but do we have to import it in every scss file?

**Review by xixas — CHANGES_REQUESTED** [review]

File: apps/portal/src/app/features/file-processor/file-processor.component.html [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      <th>Status</th>
+      <th>Action</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr *ngFor="let job of jobs">
+      <td>{{ job.timestamp }}</td>
+      <td>{{ job.fileName }}</td>
+      <td
+        [ngClass]="{
```
[the remark this answers is not in the document]
Comment (Hitesh-Kandpal) [reply apps/portal/src/app/features/file-processor/file-processor.component.html]: used tags from primeng

File: apps/portal/src/app/features/file-processor/file-processor.component.ts (hunk 1 above)
Comment (Hitesh-Kandpal) [reply apps/portal/src/app/features/file-processor/file-processor.component.ts]: status are coming from enum value now

File: apps/portal/src/app/features/file-processor/file-processor.component.ts (hunk 2 above)
Comment (Hitesh-Kandpal) [reply apps/portal/src/app/features/file-processor/file-processor.component.ts:117]: used file saver package for downloading files

File: apps/portal/src/app/features/transcript-analyzer/transcript-analyzer.component.scss (hunk 3 above)
Comment (Hitesh-Kandpal) [reply apps/portal/src/app/features/transcript-analyzer/transcript-analyzer.component.scss:1]: without importing it I am not able to use styleclass in scss file and ng server is throwing error, I tried to import it in global scss but still I am not able to use styleclass in component.scss

File: apps/portal/src/app/features/file-processor/file-processor.component.ts [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+import { HttpClient } from '@angular/common/http';
+import { JobDetails } from '../../shared/job-details.interface';
+
+@Component({
+  selector: 'app-file-processor',
+  templateUrl: './file-processor.component.html',
+  styleUrl: './file-processor.component.scss',
+})
+export class FileProcessorComponent implements OnInit {
+  jobs: JobDetails[] = [];
```
[the remark this answers is not in the document]
Comment (Hitesh-Kandpal) [reply apps/portal/src/app/features/file-processor/file-processor.component.ts:14]: added the comment

**Review by xixas — APPROVED** [review]

**Pull request merged**